packages/react/test/evaluation.spec.tsx

```diff
@@ -0,0 +1,384 @@
+      expect(screen.queryByText(FALLBACK)).toBeInTheDocument();
+      // eventually we should the value
+      await waitFor(() => expect(screen.queryByText(STATIC_FLAG_VALUE)).toBeInTheDocument(), { timeout: DELAY * 2 });
+    });
+  });
+
+  describe('suspendWhileReconciling=true (default)', () => {
+    it('should suspend until reconciled and then render', async () => {
+      await OpenFeature.setContext(SUSPENSE_OFF, {});
+      OpenFeature.setProvider(SUSPENSE_ON, suspendingProvider());
+
+      render(
+        // disable suspendUntilReady, we are only testing reconcile suspense.
+        <OpenFeatureProvider domain={SUSPENSE_ON} suspendUntilReady={false}>
+          <React.Suspense fallback={<div>{FALLBACK}</div>}>
+            <TestComponent></TestComponent>
+          </React.Suspense>
+        </OpenFeatureProvider>,
+      );
+      
+      // initially should be default, because suspendUntilReady={false}
+      expect(screen.queryByText(DEFAULT)).toBeInTheDocument();
+
+      // update the context without awaiting
+      act(() => {
+        OpenFeature.setContext(SUSPENSE_ON, { user: TARGETED_USER });
+      });
+
+      // expect to see fallback while we are reconciling
+      await waitFor(() => expect(screen.queryByText(FALLBACK)).toBeInTheDocument(), { timeout: DELAY / 2 });
+
+      // make sure we updated after reconciling
+      await waitFor(() => expect(screen.queryByText(TARGETED_FLAG_VALUE)).toBeInTheDocument(), { timeout: DELAY * 2 });
+    });
+  });
+
+  describe('suspend=false', () => {
+    it('should not suspend until reconciled and then render', async () => {
+      await OpenFeature.setContext(SUSPENSE_OFF, {});
+      OpenFeature.setProvider(SUSPENSE_OFF, suspendingProvider());
+
+      render(
+        // disable suspendUntilReady, we are only testing reconcile suspense.
+        <OpenFeatureProvider domain={SUSPENSE_OFF} suspend={false}>
+          <React.Suspense fallback={<div>{FALLBACK}</div>}>
+            <TestComponent></TestComponent>
+          </React.Suspense>
+        </OpenFeatureProvider>,
+      );
+      
+      // assert no suspense
+      expect(screen.queryByText(DEFAULT)).toBeInTheDocument();
+      expect(screen.queryByText(FALLBACK)).toBeNull();
+
+      // expect to see static value after we are ready
+      await waitFor(() => expect(screen.queryByText(STATIC_FLAG_VALUE)).toBeInTheDocument(), { timeout: DELAY * 2 });
+
+      // update the context without awaiting
+      act(() => {
+        OpenFeature.setContext(SUSPENSE_OFF, { user: TARGETED_USER });
+      });
+
+      // expect to see static value until we reconcile
+      await waitFor(() => expect(screen.queryByText(STATIC_FLAG_VALUE)).toBeInTheDocument(), { timeout: DELAY / 2 });
+
+      // make sure we updated after reconciling
+      await waitFor(() => expect(screen.queryByText(TARGETED_FLAG_VALUE)).toBeInTheDocument(), { timeout: DELAY * 2 });
+    });
+  });
+});
```

Comment:
**Suggestion**: It feels like a lot of these tests are creating `DOM` elements for the sake of verifying that the values expected to indeed evaluate from the `hook`s appropriately. Am I correct in that assumption? 

If so, I'm wondering if using the [`renderHook`](https://testing-library.com/docs/solid-testing-library/api/#renderhook) pattern to test the hooks explicitly may be a more straight forward and clear way to test these. 

The provider would still be valuable as you'd need to send that to `renderHook` as a `wrapper` component, but ultimately you could avoid having to build `TestComponent`, and variations of, in every component. 

In theory this would speed up tests over time as they'd be less concerned with rendering setup in headless `DOM`. 

Thoughts?

Reply:
Hey @therynamo , I used this for the `useOpenFeatureClient` hook tests. It's a nice pattern!